Fix Kubernetes Maven Plugin integration (`kubernetes-maven-plugin/it`) tests failing on windows

### Component

JKube Kit

### Task description

### Description

Related to https://github.com/eclipse/jkube/issues/1338

⚠️ A windows machine is required to reproduce and fix problems in this issue

When executing `mvn clean install` on Windows, the following unit tests are failing in `kubernetes-maven-plugin/it` module:

- xml-image
- metadata-labels-annotations
- xml-image-multilayer

You can run a single test in `kubernetes-maven-plugin/it` module like this:
```
mvn -Dinvoker.test=xml-image clean install
```

I think it might have to do with line endings in test expectations `\n` vs line endings in generated Dockerfile / Kubernetes manifests. If this is the case, we would need to modify [ResourceVerify](https://github.com/eclipse/jkube/blob/master/jkube-kit/common-it/src/main/java/org/eclipse/jkube/kit/common/ResourceVerify.java) class to ignore difference in line endings while comparing files.

### Expected Behavior
Kubernetes Maven Plugin integration tests should pass on windows

### Acceptance Criteria
- [ ] Kubernetes maven Plugin integration tests pass on windows

I've created https://github.com/eclipse-jkube/jkube/pull/3014 to fix `metadata-labels-annotations` integration test. It looks like it does expect `\n` inside multiline scalar values.

Regarding other two integration test failures `xml-image `and `xml-image-multilayer`, these are failing due to differences in line endings of generated Dockerfile. JKube is generating Dockerfile with Line Feed 
![jkubedockerfile-kubernetesmavenplugin-it-failure](https://github.com/eclipse-jkube/jkube/assets/13834498/381ef78a-0b16-40e3-9038-1b5d0e8c9805)

https://github.com/eclipse-jkube/jkube/blob/061c9191bcf820ff1c00b53f8479306ba7f69611/jkube-kit/config/image/src/main/java/org/eclipse/jkube/kit/config/image/build/DockerFileKeyword.java#L63

If we change this to `System.lineSeparator()` integration test start failing due to this error : 
```
[INFO] JIB> I/O error for image [registry.access.redhat.com/ubi9/ubi-minimal]:                           
                
[INFO]  
[INFO] JIB>     java.net.SocketException                                                                 
                
[INFO]  
[INFO] JIB>     Connection reset                                                                         
                
[INFO]  
[INFO] JIB> [============                  ] 38.8% complete > pulling platform-specific manifests and con
tainer configs
[INFO] [ERROR] k8s: Failed to execute the build [Error when building JIB image]
[INFO] [INFO] ------------------------------------------------------------------------
[INFO] [INFO] BUILD FAILURE
[INFO] [INFO] ------------------------------------------------------------------------
[INFO] [INFO] Total time:  3.546 s
[INFO] [INFO] Finished at: 2024-05-02T23:43:23+05:30
[INFO] [INFO] ------------------------------------------------------------------------
[INFO] [ERROR] Failed to execute goal org.eclipse.jkube:kubernetes-maven-plugin:1.17-SNAPSHOT:build (defa
ult-cli) on project xml-image-multilayer: Failed to execute the build: Error when building JIB image: Una
ble to build the image tarball: java.net.SocketException: Connection reset -> [Help 1]
[INFO] [ERROR]  
[INFO] [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[INFO] [ERROR] Re-run Maven using the -X switch to enable full debug logging.
[INFO] [ERROR]  
[INFO] [ERROR] For more information about the errors and possible solutions, please read the following ar
ticles:
[INFO] [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[INFO] run post-build script verify.groovy
[INFO] Running post-build script: /home/rokumar/work/repos/jkube/kubernetes-maven-plugin/it/target/it/xml
-image-multilayer/verify.groovy
[INFO] java.nio.file.NoSuchFileException: /home/rokumar/work/repos/jkube/kubernetes-maven-plugin/it/targe
t/it/xml-image-multilayer/target/docker/other/simple-image-test/latest/build/Dockerfile
```

I think Jib build seems to have some problem with this change.

We can stick to converting line separators to \n in ResourceVerify

https://github.com/eclipse-jkube/jkube/blob/061c9191bcf820ff1c00b53f8479306ba7f69611/jkube-kit/common-it/src/main/java/org/eclipse/jkube/kit/common/ResourceVerify.java#L91

To this: 
```java
public static String readFile(File path) throws IOException {
 return new String(FileCopyUtils.copyToByteArray(Files.newInputStream(path.toPath())), Charset.defaultCharset()).replaceAll(System.lineSeparator(), "\n");
}
```

--- a/jkube-kit/common-it/src/main/java/org/eclipse/jkube/kit/common/ResourceVerify.java
+++ b/jkube-kit/common-it/src/main/java/org/eclipse/jkube/kit/common/ResourceVerify.java
@@ -88,7 +88,7 @@ public class ResourceVerify {
   }
 
   public static String readFile(File path) throws IOException {
-    return new String(FileCopyUtils.copyToByteArray(Files.newInputStream(path.toPath())), Charset.defaultCharset());
+    return new String(FileCopyUtils.copyToByteArray(Files.newInputStream(path.toPath())), Charset.defaultCharset()).replaceAll(System.lineSeparator(), "\n");
   }
 
   public static TestContext createTestContext() {

